packages/@react-spectrum/s2/src/SegmentedControl.tsx

```diff
@@ -0,0 +1,274 @@
+}, getAllowedOverrides());
+
+const controlItem = style({
+  position: 'relative',
+  display: 'flex',
+  forcedColorAdjust: 'none',
+  color: {
+    default: 'gray-700',
+    isHovered: 'neutral-subdued',
+    isSelected: 'neutral',
+    isDisabled: 'disabled',
+    forcedColors: {
+      default: 'ButtonText',
+      isDisabled: 'GrayText',
+      isSelected: 'HighlightText'
+    }
+  },
+  // TODO: update this padding for icon-only items when we introduce the non-track style back
+  paddingX: 'edge-to-text',
+  height: 32,
+  alignItems: 'center',
+  flexBasis: 0,
+  flexGrow: 1,
+  flexShrink: 0,
+  justifyContent: 'center',
+  whiteSpace: 'nowrap',
+  '--iconPrimary': {
+    type: 'fill',
+    value: 'currentColor'
+  }
+}, getAllowedOverrides());
+
+const slider = style({
+  ...focusRing(),
+  backgroundColor: 'gray-25',
+  width: 'full',
+  height: 'full',
+  position: 'absolute',
+  boxSizing: 'border-box',
+  borderStyle: 'solid',
+  borderWidth: 2,
+  borderColor: {
+    default: 'gray-900',
+    isDisabled: 'disabled'
+  },
+  borderRadius: 'lg'
+});
+
+interface InternalSegmentedControlContextProps {
+  register?: (value: string, isDisabled?: boolean) => void,
+  prevRef?: RefObject<DOMRect | null>,
+  currentSelectedRef?: RefObject<HTMLDivElement | null>
+}
+
+interface DefaultSelectionTrackProps {
+  defaultValue?: string | null,
+  value?: string | null,
+  children?: ReactNode,
+  prevRef: RefObject<DOMRect | null>,
+  currentSelectedRef: RefObject<HTMLDivElement | null>
+}
+
+const InternalSegmentedControlContext = createContext<InternalSegmentedControlContextProps>({});
+
+function SegmentedControl(props: SegmentedControlProps, ref: DOMRef<HTMLDivElement>) {
+  [props, ref] = useSpectrumContextProps(props, ref, SegmentedControlContext);
+  let {
+    defaultValue,
+    value
+  } = props;
+  let domRef = useDOMRef(ref);
+
+  let prevRef = useRef<DOMRect>(null);
+  let currentSelectedRef = useRef<HTMLDivElement>(null);
+
+  let onChange = (value: string) => {
+    if (currentSelectedRef.current) {
+      prevRef.current = currentSelectedRef?.current.getBoundingClientRect();
+    }
+    
+    if (props.onChange) {
+      props.onChange(value);
+    }
+  };
+
+  return (
+    <RadioGroup 
+      {...props}
+      ref={domRef}
+      orientation="horizontal"
+      style={props.UNSAFE_style}
+      onChange={onChange}
+      className={(props.UNSAFE_className || '') + segmentedControl({size: 'M'}, props.styles)}
+      aria-label={props['aria-label']}>
+      <DefaultSelectionTracker defaultValue={defaultValue} value={value} prevRef={prevRef} currentSelectedRef={currentSelectedRef}>
+        {props.children}
+      </DefaultSelectionTracker>
+    </RadioGroup>
+  );
+}
+
+function DefaultSelectionTracker(props: DefaultSelectionTrackProps) {
+  let state = useContext(RadioGroupStateContext);
+  let isRegistered = useRef(!(props.defaultValue == null && props.value == null));
+  let disabledIsRegistered = useRef<string | null>(null);
```

Comment:
```suggestion
  let disabledIsRegistered = useRef<string | null>(props.defaultValue ?? props.value ?? null);
```
should probably also set this